[jewiteo] iP

jewiteo opened the pull request:
# ChatGPT
> Code is like humor. When you have to explain it, it’s bad.

ChatGPT frees your mind of having to remember things you need to do! It's,
- Text-based
- Easy to learn
- ~~Fast~~ Extremely Fast to use

All you need to do is, 
1. Pull from this [repo](https://github.com/jewiteo/ip)
2. Build and run it using gradle
3. Add your tasks.
4. Let it manage your tasks for you. :shipit: 

And best of all it is **FREE!**

Features *(so far)*:
- [x] Managing Tasks
- [x] Managing Deadlines
- [ ] Creating Reminders *(Coming Soon)*

***
If you are a Java programmer, you try changing the file name in the `main` method as seen below:

```Java
public class ChatGPT {
    public static void main(String[] args) {
        new ChatGPT("\\data\\<New File Name>").run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jewiteo committed:
Renamed and added greeting message

jewiteo committed:
Refractored code and allowed bot to echo commands by user

jewiteo committed:
added ability to store text and display when user request

jewiteo committed:
Added ability to mark task as done and change back to undone

jewiteo committed:
Added three types of task and removed default case

jewiteo committed:
Updated text-ui-test to be consistent with this project

jewiteo committed:
Added error handling to chatbot

jewiteo committed:
Added support to delete tasks

jewiteo committed:
Convert to Enums and Switch to control the commands

jewiteo committed:
Add loading of data from external text file

jewiteo committed:
Add saving of data to local text file

jewiteo committed:
Merge branch 'branch-level-7'

jewiteo committed:
Change the format of deadline and event to reflect date and time

jewiteo committed:
Merge branch 'branch-level-8'

rachelyeohm commented on src/main/java/Events.java:
```diff
@@ -0,0 +1,80 @@
+                    "/from <Start Date/Time> /to <End Date/Time>");
+        }
+        try {
+            LocalDate sDate = LocalDate.parse(startDate.split(" ")[1]);
```
Variables should be meaningfully named, so startDate might be better etc.

rachelyeohm commented on src/main/java/Deadlines.java:
```diff
@@ -0,0 +1,61 @@
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+public class Deadlines extends Task {
```
Class names could be more appropriately named. If Task is singular, then Deadline should probably be too.

rachelyeohm commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,41 @@
+public abstract class Task {
+    private String task;
+    private boolean isCompleted;
```
I like the correct use of booleans

rachelyeohm commented on src/main/java/ChatGPT.java:
```diff
@@ -0,0 +1,234 @@
+    private final static String NAME = "ChatGPT";
+    private final static String LINE = "________________________________________________";
+    private static ArrayList<Task> list = new ArrayList<Task>();
+    private enum Commands {
```
enums like this should probably be singular too - like Command

AmadeusEka commented on src/main/java/ChatGPT.java:
```diff
@@ -0,0 +1,234 @@
+        do {
+            try {
+                switch(Commands.getCommand(userInput.next().toUpperCase())) {
+                case BYE:
+                    sendExit();
+                    return;
+
+                case LIST:
+                    printList();
+                    break;
+
+                case TODO:
```
Great use of constants to increase readability of code!

AmadeusEka commented on src/main/java/ChatGPT.java:
```diff
@@ -0,0 +1,234 @@
+        File currentDir = new File(".");
+        File dataDir = new File (currentDir.getAbsolutePath() + "\\data");
+        File data = new File (dataDir.getAbsolutePath() + "\\data.txt");
+        ArrayList<Task> result = new ArrayList<>();
```
Perhaps plural form "results" can be used to represent this collection of tasks?

AmadeusEka commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,41 @@
+        this.isCompleted = isCompleted;
+    }
+
+    public String setCompleted(boolean completed) {
+        if (this.isCompleted == completed) {
+            throw new IllegalArgumentException("\t It seems the task has already been marked as such");
+        }
```
Perhaps boolean variables/methods should be named to sound like booleans, e.g. isCompleted

AmadeusEka commented on src/main/java/Events.java:
```diff
@@ -0,0 +1,80 @@
+    @Override
+    public String toPrint() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(" uuuu-MM-dd HHmm ");
+        return "event " + super.toPrint() +
+                "|" + this.startDate.format(formatter) +
+                "|" + this.endDate.format(formatter);
```
Maybe adding a line break before the operator would improve readability? According to the layout section of the coding standards.

jewiteo committed:
Refactor code to extract related code as classes

jewiteo committed:
Merge branch 'branch-A-MoreOOP'

jewiteo committed:
Add classes into suitable java packages

jewiteo committed:
Merge branch 'branch-A-Packages'

jewiteo committed:
Merge branch 'add-gradle-support'

jewiteo committed:
Refactor class to include it in the package

jewiteo committed:
Merge branch 'branch-A-Packages'

jewiteo committed:
Revert "Merge branch 'branch-A-Packages'"

This reverts commit 3a2bc7ccecec995c5b7f26627f90bc005efc355b, reversing
changes made to 479d756e8b72f9fbcb90c7d41d3ca1677905e15b.

jewiteo committed:
Set up gradle and refactor main class into a package

jewiteo committed:
Merge branch 'branch-A-Gradle'

jewiteo committed:
Add JUnit tests to test the behaviour of Parser and Storage

jewiteo committed:
Merge branch 'branch-A-JUnit'

jewiteo committed:
Update gradle to package app as Jar

jewiteo committed:
Merge branch 'branch-A-Jar'

jewiteo committed:
Add find task by searching for keyword

jewiteo committed:
Refactor code to align with coding standards

jewiteo committed:
Add javadoc comments to the methods and class members

jewiteo committed:
Merge branch 'branch-A-JavaDoc'

jewiteo committed:
Merge branch 'branch-A-CodingStandard'

jewiteo committed:
Merge branch 'branch-level-9'

jewiteo committed:
Add Checkstyle configs and refactor code according to checkstyle

jewiteo committed:
Merge branch 'branch-A-CheckStyle'

jewiteo committed:
Add basic GUI to chatbot

jewiteo committed:
Merge branch 'branch-level-10'

jewiteo committed:
Add assert to document assumptions

Switch cases within the code makes assumption that the correct input is
parsed and given.

There are thus no default case within the switches since we assume that
all inputs will correctly map to one of the possible case.

Using assert function will ensure that this assumption is correct and
will help in testing the code in fringe cases.

The asserts are deemed as true and placed as the default cases as the
assumption we are making is that the code will never reach those default
cases of those switches.

Other edge cases within the code will throw an error and be handled
within the program itself and thus assert should not be used for those
cases.

jewiteo committed:
Refactor code for better code quality

Certain methods within the class had different levels of abstraction and
did not follow SLAP. Within the commands that had to manipulate the
inputs, data processing was done together with other high level
abstraction. Loading save file error is not reflected in the GUI.

The code needs to be easily read and understood so that modifications
in the future can be easily done. Errors are also more prone to happen
to code that are not easily understandable.

Shifting the data processing within the command classes to its own
separate function allows for the code to follow SLAP and be more
succinct.

Data processing of the inputs should still be done within the command
class as it is consistent with the expected behaviour thus separating
data processing to a separate function is preferred over processing the
data in another class.

jewiteo committed:
Merge pull request #2 from jewiteo/branch-A-Assertions

Add assert to document assumptions

jewiteo committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/chatgpt/command/AddCommand.java
#	src/main/java/chatgpt/storage/Storage.java

jewiteo committed:
Merge pull request #3 from jewiteo/branch-A-CodeQuality

Refactor code for better code quality

jewiteo committed:
Add support for managing notes

There is currently no support for adding and showing notes for the
respective tasks.

Users may want to record small snippets of textual information relevant
to the task that they want to keep track of.

Adding support for managing such notes could be done through the use of
an optional flag in the respective task command to add the notes.
Another command could also be added to show the details of the specific
task and the note associated with it.

Adding another flag within the task creation commands will allow users
to easily add notes to that task. Note should only be displayed when
specified by the user to reduce visual clutter.

jewiteo committed:
Merge branch 'branch-D-notes'

jewiteo committed:
Update JavaDoc and remove redundant methods

jewiteo committed:
Update data file location and JUNIT tests

jewiteo committed:
Add support for adjustable window size for JavaFX

jewiteo committed:
Add Screenshot of the product

jewiteo committed:
Update user guide

jewiteo committed:
Fix image directory

jewiteo committed:
Update gradle configurations to be compatible with the GUI version

jewiteo committed:
Update README and add support for gitmojis
